var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -753,30 +837,20 @@ def configure_args(self):
             # Xcode 10 dropped 32-bit support
             "--disable-multilib",
             "--enable-languages={0}".format(",".join(spec.variants["languages"].value)),
-            # Drop gettext dependency
-            "--disable-nls",
+            # always add the program prefix then link to unprefixed so we don't have to
+            # deal with the semi-randomness of when they're added otherwise
+            f"--program-prefix={target_prefix}",
         ]
-
-        # Avoid excessive realpath/stat calls for every system header
-        # by making -fno-canonical-system-headers the default.
-        if self.version >= Version("4.8.0"):
-            options.append("--disable-canonical-system-headers")
-
-        # Use installed libz
-        if self.version >= Version("6"):
-            options.append("--with-system-zlib")
-
-        if "zstd" in spec:
-            options.append("--with-zstd-include={0}".format(spec["zstd"].headers.directories[0]))
-            options.append("--with-zstd-lib={0}".format(spec["zstd"].libs.directories[0]))
-
-        # Enabling language "jit" requires --enable-host-shared.
-        if "languages=jit" in spec:
-            options.append("--enable-host-shared")
+        if "+pgo" in self.spec:
+            options.append("--enable-pgo-build=lto")
+        else:
+            options.append("--disable-pgo-build")
 
         # Binutils
         if spec.satisfies("+binutils"):
             binutils = spec["binutils"].prefix.bin
+            if not os.path.isfile(binutils.ld):
+                binutils = spec["binutils"].prefix.join(self.spec.target_triple).bin
```

Comment:
Not 100% if this ever worked the right way (as in, absolute paths to the linker). We always have the right ld/as in the path, right?

Reply:
Yes, but if using an actual cross linker we need to try the cross-name in case it’s an external. We now always build the cross name and link to the unprefixed to help with this so if we require the first stage to always be built with spack this can probably come out.